test/test_runner.go

```diff
@@ -144,7 +145,13 @@ type TestRunner struct {
 	// log messages from test cases and contracts.
 	logCollection *LogCollectionHook
 
-	backend *EmulatorBackend
+	// blockchain is mainly used to obtain system-defined
+	// contracts & their exposed types
+	blockchain *emulator.Blockchain
+
+	// contracts is a mapping of contract names to their
+	// source code.
+	contracts map[string][]byte
```

Comment:
To get these, we will leverage the `flow.json` config file, which contains a relevant mapping, e.g. :
```json
"contracts": {
    "FooContract": "./contracts/FooContract.cdc",
    "ArrayUtils": "./contracts/ArrayUtils.cdc",
    "StringUtils": "./contracts/StringUtils.cdc",
    "ApprovalVoting": "./contracts/ApprovalVoting.cdc"
  },
```

Reply:
Curious why we need this new `contract` field? Isn't the `importResolver` already doing this?

Reply:
Can we update the `importResolver` to handle the `AddressLocations`?
The idea of introducing the `importResolver` was to be able to handle all imports in one place.

Reply:
Good point :+1: I think the `importResolver` can also handle `AddressLocation` imports. It will simply ignore the address part, as there no account/address at that stage. But it can still infer the source code from the `Name` field.

Reply:
Why would it only need/have the name? Can't the complete address be used?
Sorry about all the questions, I may have forgotten some of the nitty-gritty details 😄

Reply:
No worries at all, happy to answer any question. It will help me improve my understanding as well :pray: 
```cadence
import ApprovalVoting from 0x00000002535
```
The `importResolver` implementation from `flow-cli`, can get the source code for `ApprovalVoting`, just by the `Name` field of the `AddressLocation`. Basically by utilizing the `flow.json` config file. But the `flow-cli` does not have any access to the testing framework's blockchain, so there's no need to utilize the address part. This here refers only to imports done in the test script itself. Not sure if I explained it properly :sweat_smile:

Reply:
I see. My concern is that by ignoring the address, it makes it impossible to import two contracts with the same name on two different addresses.

Reply:
On second thought, it would not make much of a difference either, since for a given deployment (for tests, in this case), the `flow.json` only supports one address.

Reply:
Thanks for the explanations. I think once `importResolver` is updated, we should be good 👍

Reply:
nice!